Column addition and deletion in various tables of the plugin. (#24)
turbot/steampipe-plugin-steampipecloud: CHANGELOG.md

## v1.0.0 [2022-08-12]

_What's new?_

- New tables added
  - [steampipecloud_workspace_snapshot](https://hub.steampipe.io/plugins/turbot/steampipecloud/tables/steampipecloud_workspace_snapshot) ([#22](https://github.com/turbot/steampipe-plugin-steampipecloud/pull/22))
  - [steampipecloud_workspace_db_log](https://hub.steampipe.io/plugins/turbot/steampipecloud/tables/steampipecloud_workspace_db_log) ([#23](https://github.com/turbot/steampipe-plugin-steampipecloud/pull/23))

> _Enhancements_

Reviewer: I think we need to call out the removed columns separately in the `Breaking changes` section in CHANGELOG.md

Reviewer: For reference https://github.com/turbot/steampipe-plugin-azure/blob/main/CHANGELOG.md#v020-2021-02-18

## v0.4.0 [2022-08-19]

_What's new?_

- New tables added

>   - [steampipecloud_workspace_snapshot](https://hub.steampipe.io/plugins/turbot/steampipecloud/tables/steampipecloud_workspace_snapshot) ([#22](https://github.com/turbot/steampipe-plugin-steampipecloud/pull/22))

Reviewer 2: Can you please order these alphabetically?

  - [steampipecloud_workspace_db_log](https://hub.steampipe.io/plugins/turbot/steampipecloud/tables/steampipecloud_workspace_db_log) ([#23](https://github.com/turbot/steampipe-plugin-steampipecloud/pull/23))

_Enhancements_

- Added `process_id` column to `steampipecloud_audit_log` table. ([#24](https://github.com/turbot/steampipe-plugin-steampipecloud/pull/24))
- Added `created_by_id`, `created_by`, `updated_by_id`, `updated_by` columns to `steampipecloud_connection` table. ([#24](https://github.com/turbot/steampipe-plugin-steampipecloud/pull/24))
- Added `created_by_id`, `updated_by_id` columns to `steampipecloud_organization` table. ([#24](https://github.com/turbot/steampipe-plugin-steampipecloud/pull/24))

> - Modified columns `created_by`, `updated_by` to store additional information about the user who created or updated the organization in `steampipecloud_organization` table. ([#24](https://github.com/turbot/steampipe-plugin-steampipecloud/pull/24))

Reviewer 2: ```suggestion
- Updated columns `created_by`, `updated_by` to store additional information about the user who created or updated the organization in `steampipecloud_organization` table. ([#24](https://github.com/turbot/steampipe-plugin-steampipecloud/pull/24))
```

We consistently use `updated` more than `modified` in CHANGELOG.

> - Added `user`, `scope`, `created_by_id`, `updated_by_id` columns to `steampipecloud_organization_member` table. ([#24](https://github.com/turbot/steampipe-plugin-steampipecloud/pull/24))

Reviewer 2: When there are multiple columns, can you please order alphabetically?

- Added `identity_handle`, `identity_type`, `workspace_handle`, `connection_handle`, `connection`, `created_by_id`, `created_by`, `updated_by_id`, `updated_by` columns to `steampipecloud_workspace_connection` table. ([#24](https://github.com/turbot/steampipe-plugin-steampipecloud/pull/24))
- Added `identity_handle`, `workspace_handle`, `created_by_id`, `created_by`, `updated_by_id`, `updated_by`, `version_id` columns to `steampipecloud_workspace_mod` table. ([#24](https://github.com/turbot/steampipe-plugin-steampipecloud/pull/24))
- Added `version_id` column to `steampipecloud_workspace_mod_variable` table. ([#24](https://github.com/turbot/steampipe-plugin-steampipecloud/pull/24))

_Breaking changes_

- Removed column `email` from `steampipecloud_organization_member` table. ([#24](https://github.com/turbot/steampipe-plugin-steampipecloud/pull/24))
- Removed column `email` from `steampipecloud_organization_workspace_member` table. ([#24](https://github.com/turbot/steampipe-plugin-steampipecloud/pull/24))
- Removed column `email` from `steampipecloud_user` table. ([#24](https://github.com/turbot/steampipe-plugin-steampipecloud/pull/24))

> - Modified column `workspace_state` renamed to `state` in `steampipecloud_workspace` table. ([#24](https://github.com/turbot/steampipe-plugin-steampipecloud/pull/24))

Reviewer 2: ```suggestion
- Renamed column `workspace_state` to `state` in `steampipecloud_workspace` table. ([#24](https://github.com/turbot/steampipe-plugin-steampipecloud/pull/24))
```

I think we can shorten this note, and others like it

- Modified column `mod_constraint` renamed to `constraint` in `steampipecloud_workspace_mod` table. ([#24](https://github.com/turbot/steampipe-plugin-steampipecloud/pull/24))

> - Modified column `created_by_handle` renamed to `created_by_id` to store identifier of the person who created the setting in `steampipecloud_workspace_mod_variable` table. ([#24](https://github.com/turbot/steampipe-plugin-steampipecloud/pull/24))

Reviewer 2: ```suggestion
- Renamed column `created_by_handle` to `created_by_id` and updated the column to store the identifier of the person who created the variable in `steampipecloud_workspace_mod_variable` table. ([#24](https://github.com/turbot/steampipe-plugin-steampipecloud/pull/24))
```

Can you please update other items like this as well?

Reviewer 3: We actually store the identifier of the user who created a setting on the variable. The variables are created by steampipe cloud.

Reviewer 2: Oh ok, I see, thanks!
